packages/sit-onyx/src/components/OnyxDataGrid/features/sorting/sorting.ts

```diff
@@ -0,0 +1,68 @@
+import { computed, h, ref } from "vue";
+import { createFeature } from "..";
+import { injectI18n } from "../../../../i18n";
+import type { DataGridEntry } from "../../types";
+import SortAction from "./SortAction.vue";
+
+export type SortDirection = "asc" | "desc" | "none";
+
+export const nextSortDirection = (current?: SortDirection): SortDirection => {
+  switch (current) {
+    case "asc":
+      return "desc";
+    case "desc":
+      return "none";
+    case "none":
+    default:
+      return "asc";
+  }
+};
+
+const SORTING_FEATURE = Symbol("Sorting");
+
+export const useTableSorting = createFeature(<TEntry extends DataGridEntry>() => {
+  const sortColumn = ref<keyof TEntry>();
+  const sortDirection = ref<SortDirection>("none");
+  const locale = injectI18n().locale;
+  const intlCompare = computed(() => new Intl.Collator(locale.value).compare);
+
+  const handleClick = (column: keyof TEntry) => {
+    if (sortColumn.value === column) {
+      sortDirection.value = nextSortDirection(sortDirection.value);
+    } else {
+      sortColumn.value = column;
+      sortDirection.value = nextSortDirection();
+    }
+    if (sortDirection.value === "none") {
+      sortColumn.value = undefined;
+    }
+  };
+
+  const sortData = (data: TEntry[]) => {
+    const column = sortColumn.value;
+    if (!column || sortDirection.value === "none") {
+      return;
+    }
+    const multiplicand = sortDirection.value === "asc" ? 1 : -1;
+    data.sort((a, b) => intlCompare.value(String(a[column]), String(b[column])) * multiplicand);
```

Comment:
Sort modifies the array in place but this is not critical because we pass cloned data, right?

Reply:
Yes, for the mutations, it's even intentional :)